core/packet.c

```diff
@@ -93,6 +93,26 @@ Contains code snippets which are:
 
 #include <stdio.h>
 
+uint16_t coap_block_size = COAP_DEFAULT_BLOCK_SIZE;
+
+void lwm2m_set_block_size(uint16_t coap_block_size_arg)
+{
+	coap_block_size = coap_block_size_arg;
```

Comment:
I'd like to see the set do the validation and return a true/false indication to prevent misuse and simplify the API for the user.

Reply:
Sounds sensible. I will remove `lwm2m_validate_block_size` from the public interface then.